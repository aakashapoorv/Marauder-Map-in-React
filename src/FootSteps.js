import React, { PureComponent } from "react";
import PropTypes from "prop-types";

export default class FootSteps extends PureComponent {
  render() {
    const { name } = this.props;
    const { number } = this.props;
    const cls = `footsteps footsteps-${number}`;
    return (
      <div className={cls}>
        <div className="footstep left" />
        <div className="footstep right" />
        <div className="scroll-name">
          <p>{name}</p>
        </div>
      </div>
    );
  }
}

FootSteps.propTypes = {
  name: PropTypes.string.isRequired,
  number: PropTypes.number.isRequired
};
