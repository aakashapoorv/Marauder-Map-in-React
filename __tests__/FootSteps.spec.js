import React from "react";
import FootSteps from "../src/FootSteps";
import renderer from "react-test-renderer";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

global.shallow = shallow;
global.render = render;
global.mount = mount;

describe("FootSteps tests", () => {
  let FootStepsElement;

  beforeEach(() => {
    FootStepsElement = renderer.create(
      <FootSteps name="Harry Potter" number={2} />
    );
  });

  it("Should Render correctly", () => {
    let tree = FootStepsElement.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
