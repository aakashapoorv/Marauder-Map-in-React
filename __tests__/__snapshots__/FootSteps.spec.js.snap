// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FootSteps tests Should Render correctly 1`] = `
<div
  className="footsteps footsteps-2"
>
  <div
    className="footstep left"
  />
  <div
    className="footstep right"
  />
  <div
    className="scroll-name"
  >
    <p>
      Harry Potter
    </p>
  </div>
</div>
`;
